extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3StorageConfig.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.druid.storage.s3;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * General configurations for Amazon S3 storage.
+ */
+public class S3StorageConfig
+{
+  /**
+   * Server-side encryption type. We use a short name to match the configuration prefix with {@link S3SSEKmsConfig} and
+   * {@link S3SSECustomConfig}.
+   *
+   * @see S3StorageDruidModule#configure
+   */
+  @JsonProperty
+  private String sse = NoopServerSideEncryption.NAME;
```

Comment:
Most other places use a factory to accomplish this kind of task. Can the same be done here?

Reply:
I get your point, but not sure it's feasible for our `JsonConfigProvider` as well. Is it possible to inject an instance of a class like `ServerSideEncryptionFactory` when `JsonConfigurator` creates `S3StorageConfig`?

Reply:
The json config provider injection is... odd.

For things that are top-level-field-primitives you can specify them as a property

for things that are top-level-field-objects you can specify them as a json.

Which is not really optimal, but it works. 

so for example, it would be something like:

```
some.prefix.s3.sse={"type":"noop"}
```

Reply:
Druid's `JsonConfigProvider` is so nice! Changed to ```druid.storage.sse.type=s3```